File: read_pdfs.py
import os
from pypdf import PdfReader

pdf_dir = "public"
files = [f for f in os.listdir(pdf_dir) if f.endswith(".pdf")]

for file in files:
    try:
        reader = PdfReader(os.path.join(pdf_dir, file))
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        with open(file + ".txt", "w", encoding="utf-8") as f:
            f.write(text)
        print(f"Extracted {file}")
    except Exception as e:
        print(f"Error {file}: {e}")

File: read_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

const dir = path.join(__dirname, 'public');
const files = fs.readdirSync(dir).filter(f => f.endsWith('.pdf'));

async function run() {
    for (const file of files) {
        const dataBuffer = fs.readFileSync(path.join(dir, file));
        try {
            const data = await pdf(dataBuffer);
            fs.writeFileSync(path.join(__dirname, file + '.txt'), data.text);
            console.log(`Extracted text from ${file}`);
        } catch (err) {
            console.error(`Error reading ${file}:`, err);
        }
    }
}

console.log("pdf is", typeof pdf);
run();
